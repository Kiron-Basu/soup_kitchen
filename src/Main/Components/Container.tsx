import React from "react";

import { 
  getInitialOrderState, 
  IOrderState 
} from "../Store/OrderBundle";
import { ButtonAppBar } from "./AppBar";
import { MenuSection } from "./MenuSection";
import "../Styles/MenuSection.scss";
import "../Styles/Container.scss";

export const Container = () => {
  const defaultState: IOrderState = getInitialOrderState();
  const foodLabels: ReadonlyArray<string> = Object.keys(defaultState);
  const ref = React.createRef();
  return (
    <div className="container">
      <ButtonAppBar ref={ref} />
      <span className="container__image"></span>
      <MenuSection foodLabels={foodLabels} />
    </div>
  );
};
