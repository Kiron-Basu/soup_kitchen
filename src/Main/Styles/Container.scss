.container {
  position: fixed;
  padding: 0;
  margin: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: #f7f7f7;

  &__image {
    display: flex;
    position: relative;
    width: 350px;
    height: 150px;
    background-size: cover;
    margin: 65px auto 30px auto;
    align-items: center;
    background-image: url("../Images/logo.png");
  }
}
