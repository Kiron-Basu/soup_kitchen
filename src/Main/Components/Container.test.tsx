import React from "react";
import { 
  ShallowWrapper, 
  shallow 
} from "enzyme";

import { Container } from "./Container";

describe("App container tests", () => {
  test("Renders container correctly", () => {
    const component: ShallowWrapper = shallow(<Container />);

    expect(component).toMatchSnapshot();
  });
});
