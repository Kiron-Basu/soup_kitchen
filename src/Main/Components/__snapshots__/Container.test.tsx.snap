// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container tests Renders container correctly 1`] = `
<div
  className="container"
>
  <ForwardRef />
  <span
    className="container__image"
  />
  <MenuSection
    foodLabels={
      Array [
        "bread",
        "cheese",
        "milk",
        "butter",
        "soup",
      ]
    }
  />
</div>
`;
